lib/permissions.js

```diff
@@ -529,16 +523,57 @@ exports = module.exports = {
             }
           }
 
-          roles.forEach(function (r) {
-            r.statement.forEach(function (s, index) {
-              s.resource = s.resource.replace('{{tenantid}}', tenantId);
-              statements.push(s);
-            });
-          });
+          if (!roles || 0 === roles.length) {
+            return callback(null, []);
+          }
 
-          return callback(null, statements);
-        });
-      });
+          async.each(
+            roles,
+            function (r, nextRole) {
+              async.each(
+                r.statement,
+                function (s, nextStatement) {
+                  s.resource = s.resource.replace('{{tenantid}}', tenantId);
+                  statements.push(s);
+                  nextStatement();
+                },
+                nextRole
```

Comment:
the third argument to async.each is a callback, so just passing in the named callback `nextRole` here. If it was `nextRole()` it would be invoked inline, which would be a bug.

The alternative would be to wrap it in an anonymous callback:
````
  function (err) {
    nextRole(err);
  }
````
but it would be a bit redundant, I think. It might improve readability.